modules/govuk_elasticsearch/manifests/init.pp

```diff
@@ -142,4 +162,8 @@
   include govuk_elasticsearch::estools
 
   anchor { 'govuk_elasticsearch::end': }
+
+  $snapshots = hiera_hash('govuk_elasticsearch::snapshot::indices', {})
+  create_resources(govuk_elasticsearch::snapshot, $snapshots)
```

Comment:
Could we set this as a parameter of this class, like `$snapshot_indices`? Then we can avoid the call to `hiera_hash()` which helps to separate our data from our code and we're not using the name of a non-existent parameter on the `govuk_elasticsearch::snapshot` defined type.